packages/menu/src/Menu/utils/useMenuHeight.ts

```diff
@@ -0,0 +1,28 @@
+import { useMemo } from 'react';
+import { isUndefined } from 'lodash';
+
+import { useAvailableSpace } from '@leafygreen-ui/hooks';
+
+interface MenuHeightArgs {
+  refEl: React.RefObject<HTMLElement>;
+  spacing: number;
+  maxHeight: number;
+}
+
+export const useMenuHeight = ({
+  refEl,
+  spacing,
+  maxHeight,
+}: MenuHeightArgs) => {
+  // This hook causes a second re-render on initial load. `useAvailableSpace` uses `useViewportSize` internally, which has internal state that causes re-renders.
+  const availableSpace = useAvailableSpace(refEl, spacing);
+
+  const memoizedAvailableSpace = useMemo(
+    () => availableSpace,
+    [availableSpace],
+  );
+  const maxMenuHeightValue = !isUndefined(memoizedAvailableSpace)
+    ? `${Math.min(memoizedAvailableSpace, maxHeight)}px`
+    : 'unset';
```

Comment:
adding tests/comments would be helpful to better understand how we determine this

Reply:
I agree. Those tests should live in `useAvailableSpace` though

Reply:
Tests for that would be helpful too but original comment was with regards to this ternary with a `Math.min()`